xbmcaddon.py

```diff
@@ -225,9 +252,48 @@ def __load_add_on_xml(self):
 
         with io.open(add_on_xml, encoding='utf-8') as fp:
             xml_content = fp.read()
-            self.__version = re.findall(r'version="(\d+.\d+.\d+[^"]*)', xml_content)[0]
+            self.__version = re.findall(r'<addon.*?version="(\d+(?:\.\d)?(?:\.\d)?[^"]*)', xml_content)[0]
             self.__add_on_id = re.findall(r'addon\W+id="([^"]+)"', xml_content)[0]
             self.__name = re.findall(r'name="([^"]+)"', xml_content)[0]
+            tmp = re.findall(r'<addon.*?provider-name="([^"]+)', xml_content)
+            if tmp:
+                self.__author = tmp[0]
+
+            tmp = re.findall(r'<news>(.*?)</news>', xml_content, flags=re.DOTALL)
+            if tmp:
+                self.__news = tmp[0]
+
+            tmp = re.findall(r'<description lang=\"\w+\">(.*?)</description>', xml_content, flags=re.DOTALL)
+            if tmp:
+                for desc in tmp:
+                    lng = desc.strip('"')[1]
+                    if lng == xbmc.getLanguage():
+                        self.__description = desc
+                        break
+                else:
+                    self.__description = tmp[0]
+
+            tmp = re.findall(r'<disclaimer>(.*?)</disclaimer>', xml_content, flags=re.DOTALL)
+            if tmp:
+                self.__disclaimer = tmp[0]
+
+            tmp = re.findall(r'<fanart>(.*?)</fanart>', xml_content, flags=re.DOTALL)
```

Comment:
Does Kodi return the full path? Or perhaps the `speciall://` path? Because not it's just the relative path from the add-on main folder.